### PR TITLE
(PA-5786) Add ability to execute direct post installation scripts

### PR DESCRIPTION
**Pull request by tvpartytonight**
The existing `#add_postinstall_action` adds actions to the `%posttrans` section of an rpm spec file. This can mean that the action could fail, but the installation/upgrade could still succeed. This commit adds the ability to add actions that must succeed post installation for rpm installations by adding the action to the `%post` section.

A previous attempt to add this was done in 93ba1b4; that introduced a bug where the install and upgrade files were incorrectly deleted after the %post step. They needed to remain so that the %posttrans rpm section would detect the install/upgrade via vanagon's

Please add all notable changes to the "Unreleased" section of the CHANGELOG.

### CODE REVIEW

**Comment by tvpartytonight** [comment]
Should wait until releases are live before merging/releasing this.

**Comment by tvpartytonight** [comment]
@e-gris I think this should be good to merge and release now?

**Comment by e-gris** [comment]
Today or tomorrow, just sorting through post-release priorities.

**Review by e-gris — APPROVED** [review]

**Pull request merged**
